frontend/test/federations/federations.store.spec.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { FederationsActions } from '@/v5/store/federations/federations.redux';
+import { times } from 'lodash';
+import { selectAreStatsPending, selectContainersByFederationId, selectFavouriteFederations, selectFederationById, selectFederations, selectIsListPending } from '@/v5/store/federations/federations.selectors';
+import { ProjectsActions } from '@/v5/store/projects/projects.redux';
+import { ContainersActions } from '@/v5/store/containers/containers.redux';
+import { federationMockFactory, prepareMockContainers, prepareMockNewFederation, prepareMockSettingsReply, federationMockStats } from './federations.fixtures';
+import { createTestStore } from '../test.helpers';
+import { containerMockFactory, prepareMockViews } from '../containers/containers.fixtures';
+
+describe('Federations: store', () => {
+	let dispatch; let
+		getState;
+	const projectId = 'projectId';
+
+	beforeEach(() => {
+		({ dispatch, getState } = createTestStore());
+		dispatch(ProjectsActions.setCurrentProject(projectId));
+	});
+
+	const createAndAddFederationToStore = (federationOverrides = {}) => {
+		const mockFederation = federationMockFactory(federationOverrides);
+		dispatch(FederationsActions.fetchFederationsSuccess(projectId, [mockFederation]));
+		return mockFederation;
+	};
+
+	it('should create a federation successfully', () => {
+		createAndAddFederationToStore();
+		const mockFederation = federationMockFactory();
+		dispatch(FederationsActions.createFederationSuccess(projectId, prepareMockNewFederation(mockFederation), mockFederation._id));
+		const federationIsIncluded = selectFederationById(getState(), mockFederation._id);
+		expect(federationIsIncluded).toBeTruthy();
+	});
+
+	it('should fetch federations successfully', () => {
+		const mockFederations = times(5, () => federationMockFactory());
+		dispatch(FederationsActions.fetchFederationsSuccess(projectId, mockFederations));
+		const federations = selectFederations(getState());
```

Comment:
Just to let you know, https://github.com/3drepo/3drepo.io/pull/3532#discussion_r948155194
in this PR Santiago mentioned to test this kind of selector before populating the store. I may be worth it to add him to review this PR to see if he wants to request this change here too